src/main/java/net/tiagofar78/prisonescape/game/PrisonEscapeGame.java

```diff
@@ -567,9 +568,15 @@ public void playerMove(String playerName, PrisonEscapeLocation loc) {
         } else if (prisioner.isInRestrictedArea()) {
             prisioner.leftRestrictedArea();
         }
+    }
 
-        if (_prison.checkIfMetalDetectorTriggered(loc, player)) {
-            // TODO: Do beep
+    public void playerWalkOverMetalDetector(String playerName, PrisonEscapeLocation loc) {
+        PrisonEscapePlayer player = getPlayerOnPrisionersTeam(playerName);
+        if (player == null) {
+            return;
+        }
+        if (player.hasMetalItems()) {
+            BukkitSoundBoard.playMetalDetectorSound(loc);
```

Comment:
Use player.playSound or something like that.

Reply:
but I want it so the sound comes from the metal detector location. Not the player

Reply:
Ok, makes sense.